### PR TITLE
[DXCDT-110] [DXCDT-113] Refactor anomaly, attack protection, blacklist, custom domain, branding tests

### PR DESCRIPTION
**Pull request by sergiughf**
## References

Refactor anomaly, attack protection, blacklist, custom domain, branding tests so they can be ran in any order and they clean up the test tenant after running.

## Testing

- [x] This change adds test coverage for new/changed/fixed functionality

## Checklist

- [x] I have read and agreed to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).
- [x] I have read the [Auth0 General Contribution Guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md).
- [x] I have reviewed my own code beforehand.
- [x] I have added documentation for new/changed functionality in this PR.
- [x] All active GitHub checks for tests, formatting, and security are passing.
- [x] The correct base branch is being used, if not `main`.

### CODE REVIEW

File: management/branding_test.go [hunk 1]
```diff
@@ -3,120 +3,70 @@ package management
...
-		err = m.Branding.SetUniversalLogin(ul)
-		if err != nil {
-			t.Fatal(err)
-		}
+	actual, err := m.Branding.Read()
+	assert.NoError(t, err)
+	assert.Equal(t, expected.GetColors().GetPrimary(), actual.GetColors().GetPrimary())
+	assert.Equal(t, expected.GetFont().GetURL(), actual.GetFont().GetURL())
+	assert.Equal(t, expected.GetFaviconURL(), actual.GetFaviconURL())
+}
```
Comment (Widcket) [inline management/branding_test.go:44]: Any way to unset or cleanup those values?

File: management/attack_protection_test.go [hunk 2]
```diff
@@ -84,16 +72,15 @@ func TestAttackProtection(t *testing.T) {
...
+		assert.NoError(t, err)
 
-		t.Logf("%+v\n", suspiciousIPThrottling)
+		actual, err := m.AttackProtection.GetSuspiciousIPThrottling()
+		assert.NoError(t, err)
+		assert.Equal(t, expected.GetEnabled(), actual.GetEnabled())
+		assert.Equal(t, expected.GetStage().GetPreLogin().GetRate(), actual.GetStage().GetPreLogin().GetRate())
+		assert.Equal(t, expected.GetStage().GetPreLogin().GetMaxAttempts(), actual.GetStage().GetPreLogin().GetMaxAttempts())
+		assert.Equal(t, expected.GetStage().GetPreUserRegistration().GetRate(), actual.GetStage().GetPreUserRegistration().GetRate())
+		assert.Equal(t, expected.GetStage().GetPreUserRegistration().GetMaxAttempts(), actual.GetStage().GetPreUserRegistration().GetMaxAttempts())
```
Comment (Widcket) [inline management/attack_protection_test.go:84]: Is there any way to unset or cleanup the values updated in all the tests on this file?

**Review by Widcket — CHANGES_REQUESTED** [review]
Looks good, just left some questions regarding cleanup in some tests.

File: management/branding_test.go (hunk 1 above)
Comment (sergiughf) [reply management/branding_test.go:52]: [6fcbd2d](https://github.com/auth0/go-auth0/pull/50/commits/6fcbd2d49254b9c3c0894e89219d9bb14838da38) Good point! Thanks for spotting this!

File: management/attack_protection_test.go (hunk 2 above)
Comment (sergiughf) [reply management/attack_protection_test.go:104]: [a000577](https://github.com/auth0/go-auth0/pull/50/commits/a0005778f2c417956d5fedcb1238da20383ef46e) Absolutely!:) Thanks!

**Review by Widcket — APPROVED** [review]

**Review by willvedd — APPROVED** [review]

**Pull request merged**